### DISCUSSION TITLE
Usage of job status check functions in conditionals

### DISCUSSION THREAD

**Discussion by rajivshah3** [community/community#26881] [Actions]
<p>In my workflow, I want to be able to clean up the macOS keychain regardless of whether the job succeeded:</p>
<pre><code class="lang-yaml">   - name: Delete keychain (macOS)
     run: security delete-keychain signing.keychain
</code></pre>
<p>From the <a href="https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#always" rel="noopener nofollow ugc">documentation</a>, I can do this so that this step always runs:</p>
<pre><code class="lang-yaml">   - name: Delete keychain (macOS)
     run: security delete-keychain signing.keychain
     if: ${{ always() }}
</code></pre>
<p>However, my workflow is also run on Windows and Ubuntu, so I only want to run this step on macOS:</p>
<pre><code class="lang-yaml">    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ubuntu-latest, macos-latest, windows-latest]
    ...

   - name: Delete keychain (macOS)
     run: security delete-keychain signing.keychain
     if: matrix.os == 'macos-latest' &amp;&amp; ${{ always() }}
</code></pre>
<p>Despite the <code>matrix.os == 'macos-latest'</code> condition, this step still runs on Windows and Ubuntu. Is there a way I can always run this command on macOS only?</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="rajivshah3" data-post="1" data-topic="132950">
<div class="title">
<div class="quote-controls"></div>
 rajivshah3:</div>
<blockquote>
<pre><code class="lang-yaml">   - name: Delete keychain (macOS)
     run: security delete-keychain signing.keychain
     if: matrix.os == 'macos-latest' &amp;&amp; ${{ always() }}
</code></pre>
</blockquote>
</aside>
<p>This way only <code>always()</code> is treated as an expression, and the rest as a literal string. You need to enclose the whole expression in <code>${{ ...  }}</code>:</p>
<pre><code class="lang-yaml">   - name: Delete keychain (macOS)
     run: security delete-keychain signing.keychain
     if: ${{ matrix.os == 'macos-latest' &amp;&amp; always() }}
</code></pre>

**Comment by rajivshah3** [comment]
<p>That worked, thanks so much!</p>

**Comment by Simran-B** [comment]
<p>You don’t actually need <code>${{ }}</code> around the expression for if fields:</p>
<blockquote>
<h4><a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsif" rel="noopener nofollow ugc"> <strong><code>jobs.&lt;job_id&gt;.steps.if</code></strong> </a></h4>
<p>[…]</p>
<p>When you use expressions in an  <code>if</code>  conditional, you may omit the expression syntax ( <code>${{ }}</code> ) because GitHub automatically evaluates the  <code>if</code>  conditional as an expression</p>
</blockquote>
